### PR TITLE
Refactor Detectron2 interface

### PR DESCRIPTION
**Pull request by epwalsh**
Still TODO:
- [ ] Verify these architectures can run: https://github.com/facebookresearch/detectron2/blob/master/MODEL_ZOO.md#rpn--fast-r-cnn
- [ ] Remove custom stuff.

### CODE REVIEW

File: allennlp/modules/vision/region_detector.py [hunk 1]
```diff
@@ -61,103 +62,63 @@ class FasterRcnnRegionDetector(RegionDetector):
...
 
-        predictions = self._model.roi_heads.box_predictor(pooled_features)
+        predictions = self.model.roi_heads.box_predictor(pooled_features)
 
         # class probability
         cls_probs = F.softmax(predictions[0], dim=-1)
         cls_probs = cls_probs[:, :-1]  # background is last
 
-        predictions, r_indices = self._model.roi_heads.box_predictor.inference(
+        predictions, r_indices = self.model.roi_heads.box_predictor.inference(
```
Comment (dirkgr) [inline allennlp/modules/vision/region_detector.py:121]: This is slightly different from how detectron calls itself.

**Comment by dirkgr** [comment]
Is this ready for a look yet?

**Comment by epwalsh** [comment]
No, I'm playing around with torchvision as an alternative still.

**Comment by epwalsh** [comment]
This would be closed by https://github.com/allenai/allennlp/pull/4821.

File: allennlp/modules/vision/region_detector.py [hunk 2]
```diff
@@ -52,168 +52,74 @@ def forward(
...
+        # so to save memory we remove it from the model and patch it with a small class
+        # that will work in its place.
+        class BackbonePatch(NamedTuple):
+            size_divisibility: int = 0
 
-            pipeline = detectron.get_pipeline_from_flat_parameters(
-                self.flat_parameters, make_copy=False
+        backbone_patch = BackbonePatch(size_divisibility=model.backbone.size_divisibility)
+        del model.backbone
+        model.backbone = backbone_patch
```
Comment (dirkgr) [inline allennlp/modules/vision/region_detector.py:79]: Why is this necessary?

**Comment by dirkgr** [comment]
@epwalsh, do we still need this?
